spring-data-mongodb/src/main/java/org/springframework/data/mongodb/observability/DefaultMongoHandlerObservationConvention.java

```diff
@@ -54,8 +95,12 @@ public KeyValues getLowCardinalityKeyValues(MongoHandlerContext context) {
 	@Override
 	public KeyValues getHighCardinalityKeyValues(MongoHandlerContext context) {
 
-		return KeyValues.of(
-				HighCardinalityCommandKeyNames.MONGODB_COMMAND.withValue(context.getCommandStartedEvent().getCommandName()));
+		return KeyValues.of(HighCardinalityCommandKeyNames.MONGODB_COMMAND.withValue(context.getCommandName()));
+	}
+
+	@Override
+	public String getContextualName(MongoHandlerContext context) {
+		return context.getContextualName();
```

Comment:
It would be better to just define the contextual name here and not take it from the context.